lib/postProcessing/filter.js

```diff
@@ -46,30 +48,54 @@ filter.action = function(request, response, callback) {
   return callback();
 };
 
-filter._filterMatches = function(textToMatch, propertyText) {
-  if (textToMatch[0] === ">") {
-    textToMatch = textToMatch.substring(1);
-    if (typeof propertyText === "number") textToMatch = parseInt(textToMatch, 10);
-    if (textToMatch < propertyText) return true;
-  } else if (textToMatch[0] === "<") {
-    textToMatch = textToMatch.substring(1);
-    if (typeof propertyText === "number") textToMatch = parseInt(textToMatch, 10);
-    if (textToMatch > propertyText) return true;
-  } else if (textToMatch[0] === "~") {
-    if ((textToMatch.substring(1) + "").toLowerCase() === (propertyText + "").toLowerCase()) return true;
-  } else if (textToMatch[0] === ":") {
-    if ((propertyText + "").toLowerCase().indexOf((textToMatch.substring(1) + "").toLowerCase()) !== -1) return true;
-  } else if (textToMatch === propertyText) return true;
+filter._splitFilterElement = function(filterElementStr) {
+  if (FILTER_OPERATORS.indexOf(filterElementStr[0]) !== -1) {
+    return { operator: filterElementStr[0], value: filterElementStr.substring(1) };
+  }
+  return { operator: null, value: filterElementStr };
+};
+
+filter._filterMatches = function(filterElementStr, attributeValue, attributeConfig) {
+  var filterElement = filter._splitFilterElement(filterElementStr);
+  var validationResult = attributeConfig.validate(filterElement.value);
+  if (validationResult.error) {
+    debug.filter("invalid filter condition value:", validationResult.error);
+    return false;
+  }
+  filterElement.value = validationResult.value;
+  if (!filterElement.operator) {
+    return attributeValue === filterElement.value;
```

Comment:
The direct comparison for non-string types (such as dates) does not appear to be working.

I'd expect the following URL to return a single record titled 'How to AWS'

http://localhost:16006/rest/articles?filter[created]=2016-02-08